packages/core/src/storage/vectorStore/AstraDBVectorStore.ts

```diff
@@ -189,19 +199,25 @@ export class AstraDBVectorStore implements VectorStore {
 
     await cursor.forEach(async (row: Record<string, any>) => {
       const {
-        [this.idKey]: id,
-        [this.contentKey]: content,
         $vector: embedding,
         $similarity: similarity,
-        ...metadata
+        [this.idKey]: id,
+        [this.contentKey]: content,
+        [this.metadataKey]: metadata = {},
+        ...rest
       } = row;
 
-      const node = new Document({
-        id_: id,
-        text: content,
-        metadata,
-        embedding,
-      });
+      if (!metadata?.["_node_content"]) {
+        metadata["_node_content"] = JSON.stringify({
+          id,
+          text: content,
+          metadata,
+          ...rest,
+        });
+      }
```

Comment:
If that works, we're ready to merge :)

Reply:
Hey @marcusschiesser the problem I had w/o setting the `_node_content` was that if the vector table was created w/o llama the `metadataDictToNode` throws and error for having an empty `_node_content`

Reply:
This was the situation I was trying to solve for [here](https://github.com/run-llama/LlamaIndexTS/pull/485#discussion_r1473417942)

Reply:
@mfortman11 yes, my guess is you have been running in this error, because you have been reading data that was stored previously before adding `metadataDictToNode`.

To support this case, I moved your fallback check to the `metadataDictToNode` in this branch: https://github.com/run-llama/LlamaIndexTS/compare/mfortman11/astra?expand=1

Can you please check and test that branch? if it works, we're good to go